cadence/contracts/bridge/FlowEVMBridgeConfig.cdc

```diff
@@ -252,6 +252,38 @@ contract FlowEVMBridgeConfig {
             ?? panic("Missing or mis-typed CadenceBlocklist in storage")
     }
 
+    /// Sets the pause status of a given type, reverting if the type has no associated EVM address as either bridge-
+    /// defined or registered as a custom cross-VM association
+    ///
+    access(self)
+    fun updatePauseStatus(_ type: Type, pause: Bool) {
+        var evmAddress = ""
+        if let customAssoc = FlowEVMBridgeCustomAssociations.getEVMAddressAssociated(with: type) {
+            pause ? FlowEVMBridgeCustomAssociations.pauseCustomConfig(forType: type)
+                : FlowEVMBridgeCustomAssociations.unpauseCustomConfig(forType: type)
+            evmAddress = customAssoc.toString()
+        }
+        if let bridgedAssoc = &FlowEVMBridgeConfig.registeredTypes[type] as &TypeEVMAssociation? {
+            // no-op if already paused
+            if pause && !bridgedAssoc.isPaused {
+                bridgedAssoc.pause()
+                if evmAddress.length == 0 {
+                    // assign as bridge association only if custom association does not exist
+                    evmAddress = bridgedAssoc.evmAddress.toString()
+                }
+            } else if !pause && bridgedAssoc.isPaused {
+                bridgedAssoc.unpause()
+                if evmAddress.length == 0 {
+                    // assign as bridge association only if custom association does not exist
+                    evmAddress = bridgedAssoc.evmAddress.toString()
+                }
+            }
+        }
+        assert(evmAddress.length == 0,
```

Comment:
I'm confused. it looks like with the current logic, this assertion will fail almost every time. Should it be `evmAddress.length != 0`?

Reply:
Corrected in [118f291](https://github.com/onflow/flow-evm-bridge/pull/176/commits/118f291a8434443f4b5662ddf09635bfdb31b786)